Frends.Kafka.Consume/Frends.Kafka.Consume.Tests/MessageProducer.cs

```diff
@@ -0,0 +1,349 @@
+﻿using Avro;
+using Avro.Generic;
+using Confluent.Kafka;
+using Confluent.SchemaRegistry;
+using Confluent.SchemaRegistry.Serdes;
+using Frends.Kafka.Consume.Definitions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Text;
+
+#pragma warning disable CS8600,CS8603,CS8601,CS8604, CS8602
+namespace Frends.Kafka.Consume.Tests
+{
+    internal class MessageProducer
+    {
+        internal static async Task ProduceBasic(Input input, Options options, Socket socket, Sasl sasl, Ssl ssl, string? msgKey, string? msg, CancellationToken cancellationToken)
+        {
+            try
+            {
+                using var producer = new ProducerBuilder<string, string>(GetProducerConfig(input, options, socket, sasl, ssl)).Build();
+
+                TopicPartition topicPartition;
+                if (input.Partition >= 0)
+                    topicPartition = new TopicPartition(input.Topic, new Partition(input.Partition));
+                else
+                    topicPartition = new TopicPartition(input.Topic, new Partition());
+
+                Message<string, string> message = new()
+                {
+                    Key = msgKey ?? null,
+                    Value = msg
+                };
+
+                await producer.ProduceAsync(topicPartition, message, cancellationToken);
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+        }
+
+        internal static async Task ProduceAvro(Input input, Options options, Socket socket, Sasl sasl, Ssl ssl, string? recordJson, string? schema, string? msgKey, SchemaRegistry schemaRegistry, CancellationToken cancellationToken)
+        {
+            var schemaRegistryConfig = new SchemaRegistryConfig()
+            {
+                Url = AssignIfNotNullOrEmpty(schemaRegistry.SchemaRegistryUrl, string.Empty),
+                BasicAuthCredentialsSource = GetBasicAuthCredentialsSource(schemaRegistry.BasicAuthCredentialsSource),
+                BasicAuthUserInfo = AssignIfNotNullOrEmpty(schemaRegistry.BasicAuthUserInfo, string.Empty),
+                RequestTimeoutMs = schemaRegistry.RequestTimeoutMs,
+                MaxCachedSchemas = schemaRegistry.MaxCachedSchemas,
+            };
+
+            if (!string.IsNullOrEmpty(schemaRegistry.SslCaLocation))
+            {
+                schemaRegistryConfig.SslCaLocation = AssignIfNotNullOrEmpty(schemaRegistry.SslCaLocation, string.Empty);
+                schemaRegistryConfig.SslKeystorePassword = AssignIfNotNullOrEmpty(schemaRegistry.SslKeystorePassword, string.Empty);
+                schemaRegistryConfig.SslKeystoreLocation = AssignIfNotNullOrEmpty(schemaRegistry.SslKeystoreLocation, string.Empty);
+            }
+
+            TopicPartition topicPartition = null;
+            if (input.Partition >= 0)
```

Comment:
Is the `else` missing here?..

Reply:
Ah, I see that this is done below. Can we do the same logic for this as in ProduceBasic? It is a bit confuding why in ProductBasic it is done one way and here you do another way.